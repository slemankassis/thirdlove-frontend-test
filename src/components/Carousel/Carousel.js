import React from 'react';
import ImageGallery from 'react-image-gallery';

import './style_carousel.scss';

class Carousel extends React.Component {
  render() {
    return (
      <ImageGallery items={this.props.images} />
    );
  }
}

export default Carousel;
