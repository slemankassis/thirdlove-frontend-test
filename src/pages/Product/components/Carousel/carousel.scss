@import "~react-image-gallery/styles/scss/image-gallery.scss";
@import "./../../../../styles/variables.scss";

$ig-grey: #888888 !default;

.image-gallery {
}

.image-gallery-right-nav,
.image-gallery-left-nav,
.image-gallery-left-nav,
.image-gallery-fullscreen-button,
.image-gallery-play-button {
  display: none;
}

.image-gallery-thumbnails-wrapper {
  display: none;
}

// @media (min-width: $breakpoint-desktop) {
.image-gallery-thumbnails-wrapper {
  display: inline-block;
}

.image-gallery-thumbnails-container {
  display: flex;
  flex-direction: column;
}

.image-gallery-content {
  display: flex;
  flex-direction: row-reverse;
  margin-top: 68px;
  position: absolute;
}

.image-gallery-image {
  height: 726px;
  width: 652px;
  position: relative;
  scroll-margin-left: 241px;
}

.image-gallery-thumbnails-container {
  align-items: center;

  .image-gallery-thumbnail {
    width: 74px;
    height: 82px;
  }
}
