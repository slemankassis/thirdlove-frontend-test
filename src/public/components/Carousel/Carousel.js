const React = require('react');
const ImageGallery = require('react-image-gallery').default;

require('./style_carousel.scss');

class Carousel extends React.Component {
  render() {
    return (
      <ImageGallery items={this.props.images} />
    );
  }
}

module.exports = Carousel;
