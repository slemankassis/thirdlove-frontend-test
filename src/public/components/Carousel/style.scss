@import "~react-image-gallery/styles/scss/image-gallery.scss";

@import "./styles/image-gallery.scss";

$ig-grey: #888888 !default;

.image-gallery-slide-wrapper {
  &.left {
    width: auto;
  }
}

.image-gallery-thumbnail {
  &.active {
    border: 1px solid $ig-grey;

    @media (max-width: $ig-screen-sm-min) {
      border: 3px solid $ig-grey;
    }
  }
}

.image-gallery-bullets {
  .image-gallery-bullet {
    background: #cccccc;
    box-shadow: none;
    border: none;
    padding: 4px;

    &.active {
      background: #444444;
    }
  }
}

@media only screen and (min-width: 980px) {
  .image-gallery-slide-wrapper { // Desktop
    &.left {
      max-width: 652px;
      max-height: 726px;
      width: calc(100% - 113px);
    }
  }
}
