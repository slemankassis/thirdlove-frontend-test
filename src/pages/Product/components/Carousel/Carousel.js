import React from 'react';
import PropTypes from 'prop-types';
import ImageGallery from 'react-image-gallery';

import './carousel.scss';

class Carousel extends React.Component {
  render() {
    // TODO: Filter images by color
    return (
      <ImageGallery items={this.props.images} />
    );
  }
}

Carousel.propTypes = {
  images: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Carousel;
