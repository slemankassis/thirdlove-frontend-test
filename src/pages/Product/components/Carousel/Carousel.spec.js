import React from 'react';
import Carousel from './Carousel';

describe('Carousel', () => {
  it('renders snapshot', () => {
    const props = {
      images: mockData.product.images,
    };

    const component = shallow(<Carousel {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
