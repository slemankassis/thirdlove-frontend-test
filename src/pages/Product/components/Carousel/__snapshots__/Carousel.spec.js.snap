// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders snapshot 1`] = `"<ImageGallery items={{...}} showNav={true} autoPlay={false} lazyLoad={false} infinite={true} showIndex={false} showBullets={false} showThumbnails={true} showPlayButton={true} showFullscreenButton={true} disableThumbnailScroll={false} disableArrowKeys={false} disableSwipe={false} useTranslate3D={true} isRTL={false} useBrowserFullscreen={true} preventDefaultTouchmoveEvent={false} flickThreshold={0.4} stopPropagation={false} indexSeparator=' / ' thumbnailPosition='bottom' startIndex={0} slideDuration={450} swipingTransitionDuration={0} slideInterval={3000} swipeThreshold={30} renderLeftNav={[Function: renderLeftNav]} renderRightNav={[Function: renderRightNav]} renderPlayPauseButton={[Function: renderPlayPauseButton]} renderFullscreenButton={[Function: renderFullscreenButton]} />"`;
